odl/space/pspace.py

```diff
@@ -650,11 +650,41 @@ def __getitem__(self, indices):
 
     def __setitem__(self, indices, values):
         """Implement ``self[indices] = values``."""
+        # Get the parts to which we assign values
+        if isinstance(indices, Integral):
+            indexed_parts = (self.parts[indices],)
+            values = (values,)
+        elif isinstance(indices, slice):
+            indexed_parts = self.parts[indices]
+        elif isinstance(indices, list):
+            indexed_parts = tuple(self.parts[i] for i in indices)
+        else:
+            raise TypeError('bad index type {}'.format(type(indices)))
```

Comment:
Question here: This is more restrictive than `__getitem__` which happily iterates over anything that allows it, including tuples. That's not Numpy-style behavior -- indexing with tuples means something very different there. (They also have indexing with arrays, but that's not applicable here)